File: Python/1. Matrix Diagonal Sum/sum.py
matrix = [[1,2,3], [4,5,6], [7,8,9]]
print(matrix)

sum = 0

for i in range(len(matrix)):
    sum = sum + matrix[i][i]
    sum = sum + matrix[i][len(matrix)-1-i]
    
if len(matrix) %2 == 1:
    sum = sum - matrix[len(matrix)//2][len(matrix)//2]
    print(sum)
else:
    print(sum)

File: Java/1. Matrix Diagonal Sum/sum.java

public class sum {
    public static void main(String[] args) {
        int[][] matrix = {
                { 1, 2, 3 },
                { 4, 5, 6 },
                { 7, 8, 9 }
        };

        int diagonalSum = calculateDiagonalSum(matrix);
        System.out.println("Diagonal sum: " + diagonalSum);
    }

    public static int calculateDiagonalSum(int[][] matrix) {
        int sum = 0;
        int rows = matrix.length;
        int cols = matrix[0].length;

        if (rows != cols) {
            System.out.println("The matrix is not square.");
            return 0;
        }

        for (int i = 0; i < rows; i++) {
            sum += matrix[i][i];
        }

        return sum;
    }
}
